### PR TITLE
add proper healthcheck, sd_notify and watchdog support

### PR DESCRIPTION
**Pull request by gdamjan-h4**
… also use go modules

### CODE REVIEW

**Review by hfsd — COMMENTED** [review]
Seems OK.

**Comment by HeavyHorst** [comment]
LGTM, thx!

File: main.go [hunk 1]
```diff
@@ -32,12 +36,47 @@ func init() {
...
+		} else {
+			return
+		}
+	} else {
+		return
+	}
+
+	go func(uno *unoconv, watchdog_usec time.Duration) {
+		for {
+			time.Sleep(watchdog_usec / 2)
```
Comment (HeavyHorst) [inline main.go:59]: Is it necessary to call the watchdog every x microseconds? 
Seems to me to be quite frequent.

Comment (gdamjan-h4) [reply main.go:59]: WATCHDOG_USEC is configured in the systemd unit in seconds with 
https://www.freedesktop.org/software/systemd/man/systemd.service.html#WatchdogSec=

Comment (HeavyHorst) [reply main.go:59]: Ah, ok. And systemd then sets the environment variable WATCHDOG_USEC. Thanks for the clarification!

**Pull request merged**